avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+
+    def is_mounted(self):
+        """
+        Check the NFS is mounted or not.
+
+        :return: If the src is mounted as expect
+        :rtype: Boolean
+        """
+        return utils_misc.is_mounted(self.mount_src, self.mount_dir, "nfs")
+
+    def mount(self):
+        """
+        Mount source into given mount point.
+        """
+        return utils_misc.mount(self.mount_src, self.mount_dir, "nfs",
+                                perm=self.mount_options)
+
+    def umount(self):
+        """
+        Umount the given mount point.
+        """
+        return utils_misc.umount(self.mount_src, self.mount_dir, "nfs")
+
+    def setup(self):
+        """
+        Setup NFS in host.
+
+        Mount NFS as configured. If a local nfs is requested, setup the NFS
+        service and exportfs too.
+        """
+        if self.nfs_setup:
+            if not self.nfs_service.status():
+                logging.debug("Restart NFS service.")
+                self.nfs_service.restart()
+
+            if not os.path.isdir(self.export_dir):
+                os.makedirs(self.export_dir)
+            self.exportfs.export()
+            self.unexportfs_in_clean = not self.exportfs.already_exported
+
+        logging.debug("Mount %s to %s" % (self.mount_src, self.mount_dir))
+        if os.path.exists(self.mount_dir):
+            if not os.path.isdir(self.mount_dir):
+                raise OSError(
+                    "Mount point %s is not a directory, check your setup." %
+                    self.mount_dir)
+
+        if not os.path.isdir(self.mount_dir):
+            os.makedirs(self.mount_dir)
+            self.mk_mount_dir = True
+        self.mount()
+
+    def cleanup(self):
+        """
+        Clean up the host env.
+
+        Umount NFS from the mount point. If there has some change for exported
+        file system in host when setup, also clean up that.
+        """
+        self.umount()
+        if self.nfs_setup and self.unexportfs_in_clean:
+            self.exportfs.reset_export()
+        if self.mk_mount_dir and os.path.isdir(self.mount_dir):
+            utils_misc.safe_rmdir(self.mount_dir)
+
+
+class NFSClient(object):
+
+    """
+    NFSClient class for handle nfs remotely mount and umount.
+    """
+
+    def __init__(self, params):
+        # Setup SSH connection
+        print params
+        self.ssh_obj = SSHConnection(params)
+        ssh_timeout = int(params.get("ssh_timeout", 10))
+        self.ssh_obj.conn_setup(timeout=ssh_timeout)
+
+        self.mkdir_mount_remote = False
+        self.mount_dir = params.get("nfs_mount_dir")
+        self.mount_options = params.get("nfs_mount_options")
+        self.mount_src = params.get("nfs_mount_src")
+        self.nfs_client_ip = params.get("nfs_client_ip")
+        self.nfs_server_ip = params.get("nfs_server_ip")
+        self.ssh_user = params.get("ssh_username", "root")
+        self.remote_nfs_mount = params.get("remote_nfs_mount", "yes")
+
+    def is_mounted(self):
```

Comment:
With the `process.run`-like interface this could simply use the same code as the `utils_misc` (resp. the util I suggested to call `avocado.utils.mount`) if you make it also to (optionally) accept `process.run`-like object.